client/logmon/logmon.go

```diff
@@ -211,24 +223,24 @@ func newLogRotatorWrapper(path string, logger hclog.Logger, rotator *logging.Fil
 		openCompleted:     make(chan struct{}),
 		logger:            logger,
 	}
-	wrap.start(fifoOpenFn)
+
+	wrap.start(openFn)
 	return wrap, nil
 }
 
 // start starts a goroutine that copies from the pipe into the rotator. This is
 // called by the constructor and not the user of the wrapper.
-func (l *logRotatorWrapper) start(readerOpenFn func() (io.ReadCloser, error)) {
+func (l *logRotatorWrapper) start(openFn func() (io.ReadCloser, error)) {
 	go func() {
 		defer close(l.hasFinishedCopied)
 
-		reader, err := readerOpenFn()
+		reader, err := openFn()
+		close(l.openCompleted)
 		if err != nil {
-			close(l.openCompleted)
-			l.logger.Warn("failed to open log fifo", "error", err)
+			l.logger.Warn("failed to open fifo", "error", err)
 			return
 		}
 		l.processOutReader = reader
-		close(l.openCompleted)
```

Comment:
`l.openCompleted` channel is currently synchronizing access for `l.processOutReader` which is accessed in a goroutine in https://github.com/hashicorp/nomad/blob/f3652ee95bc80cf5cb0d1750e8ab9c85ed5febe6/client/logmon/logmon.go#L278-L284 .

Reply:
oh yep - this was a totally accidental change as I was reverting some other stuff.